### PR TITLE
Suggest config to nevergrad

### PR DESCRIPTION
**Pull request by chaosagent**
Make hand_coded_optimizations output a suggestion for nevergrad. Currently does not handle group_for_reduce or WMMA.

### CODE REVIEW

File: tinygrad/ops.py [hunk 1]
```diff
@@ -167,7 +167,8 @@ def exec_ast(self, ast:LazyOp, output, **kwargs):
     if output.realized:
       if output.realized.__class__ is RawConst: output.realized = None  # can't assign to RawConst
       for a in ast.buffers:
-        if a.realized == output.realized and not a.st.contiguous:
+        # TODO: KOPT does not yet saving and restoring buffers
```
Comment (wozeparrot) [inline tinygrad/ops.py:170]: does not support?

File: tinygrad/codegen/optimizer.py [hunk 2]
```diff
@@ -166,6 +201,7 @@ def hand_coded_optimizations(k:Linearizer):
       # TODO: use 1024 if it's allowed in a smarter way
       for sz in (([256, 16]) if prod(k.sts[0].shape[:k.first_reduce]) <= 32 else [16]):
         if all(st.shape[k.first_reduce] % sz == 0 or st.shape[k.first_reduce] == 1 for st in k.sts):
+          # suggestion['T', k.axis_idxs[k.first_reduce]] = sz
```
Comment (wozeparrot) [inline tinygrad/codegen/optimizer.py:204]: intended comment? same below

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/codegen/linearizer.py             5               0
tinygrad/codegen/optimizer.py             61              25
tinygrad/ops.py                            2               1
------------------------------------------------------------
total                                     68              26
------------------------------------------------------------
lines added in the tinygrad folder: 42
```

**Comment by chaosagent** [comment]
I think I can do better.

**Pull request closed without merging**